tests/flows/fulfillment/test_configuration.py

```diff
@@ -0,0 +1,409 @@
+
+    mocked_next_step.assert_not_called()
+
+
+def test_subscription_update_auto_renewal_step_all_failed(
+    mocker,
+    order_factory,
+    subscriptions_factory,
+):
+    """
+    Test the case when all subscription updates fail.
+    """
+    subscriptions = subscriptions_factory()
+    order = order_factory(
+        subscriptions=subscriptions,
+    )
+
+    mocked_adobe_client = mocker.MagicMock()
+    mocked_adobe_client.get_subscription.return_value = None
+    mocker.patch(
+        "adobe_vipm.flows.fulfillment.configuration.get_adobe_client",
+        return_value=mocked_adobe_client,
+    )
+
+    mocked_notify = mocker.patch(
+        "adobe_vipm.flows.fulfillment.configuration.notify_not_updated_subscriptions"
+    )
+    mocked_switch_to_failed = mocker.patch(
+        "adobe_vipm.flows.fulfillment.configuration.switch_order_to_failed"
+    )
+
+    mocked_client = mocker.MagicMock()
+    mocked_next_step = mocker.MagicMock()
+
+    context = Context(order=order, product_id='PRD-1111-1111')
+
+    step = SubscriptionUpdateAutoRenewal()
+    step(mocked_client, context, mocked_next_step)
+
+    mocked_adobe_client.get_subscription.assert_called_once_with(
+        order["authorization"]["id"],
+        '',
+        subscriptions[0]["externalIds"]["vendor"],
+    )
+    mocked_adobe_client.update_subscription.assert_not_called()
+
+    mocked_notify.assert_called_once_with(
+        order["id"],
+        f"Error updating Adobe subscription {subscriptions[0]["externalIds"]["vendor"]}: "
+        f"No matching Adobe subscription found",
+        [],
+        context.product_id,
+    )
+    mocked_switch_to_failed.assert_called_once_with(
+        mocked_client,
+        order,
+        f"Error updating Adobe subscription {subscriptions[0]["externalIds"]["vendor"]}: "
+        f"No matching Adobe subscription found",
+    )
+
+    mocked_next_step.assert_not_called()
+
+
+def test_subscription_update_auto_renewal_step_rollback_on_partial_failure(
+    mocker,
+    order_factory,
+    subscriptions_factory,
+    adobe_subscription_factory,
+):
+    """
+    Test rollback is triggered when one subscription update succeeds and a subsequent one fails.
+    """
+
+    subscriptions1 = subscriptions_factory()
+    subscriptions2 = subscriptions_factory()
+    subscriptions1[0]["externalIds"]["vendor"] = "a-sub-id_1"
+    subscriptions2[0]["externalIds"]["vendor"] = "a-sub-id_2"
+    subscriptions = [subscriptions1[0], subscriptions2[0]]
+
+    order = order_factory(subscriptions=subscriptions)
+
+    adobe_sub_1 = adobe_subscription_factory(
+        subscription_id="a-sub-id_1",
+        renewal_quantity=10,
+        autorenewal_enabled=False,
+    )
+    adobe_sub_2 = adobe_subscription_factory(
+        subscription_id="a-sub-id_2",
+        renewal_quantity=5,
+        autorenewal_enabled=False,
+    )
+
+    mocked_adobe_client = mocker.MagicMock()
+
+    def get_subscription_side_effect(auth_id, cust_id, vendor_id):
```

Comment:
that's not how it should be defined, you can define a list of side effects here, they are going to be picked up one by one for each call